pipeline-model-extensions/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/withscript/WithScriptDescriptor.java

```diff
@@ -83,4 +101,18 @@ public T newInstance(Map<String,Object> arguments) throws Exception {
     public T newInstance() throws Exception {
         return clazz.newInstance();
     }
+
+    @Extension
+    public static class WithScriptAllowlist extends GroovySourceFileAllowlist {
```

Comment:
This implementation of `GroovySourceFileAllowlist` is used for implementations of `DeclarativeAgent` and `DeclarativeStageConditional` (aka `when` conditions). `registerScript` is called by the `WithScriptDescriptor` constructor, so it automatically handles most implementations (barring a few special cases like abstract Groovy scripts), even those in other plugins like `docker-workflow` and `kubernetes`.